integration/tests/cook/util.py

```diff
@@ -455,14 +456,25 @@ def make_temporal_uuid():
 def job_label():
     return os.getenv('COOK_TEST_JOB_LABEL')
 
+def get_caller():
+    """Get the name of the function that called the caller of this function."""
+    startFrame = sys._getframe(2)
+    while startFrame is not None:
+        name = startFrame.f_code.co_name
+        if name == '<listcomp>':
+            pass
+        else:
+            return startFrame.f_code.co_name
+        startFrame = startFrame.f_back
+    return ""
 
 def minimal_job(**kwargs):
     job = {
         'command': 'echo Default Test Command',
         'cpus': get_default_cpus(),
         'max_retries': 1,
         'mem': int(os.getenv('COOK_DEFAULT_JOB_MEM_MB', 256)),
-        'name': 'default_test_job',
+        'name': ('default_test_job-'+get_caller()),
```

Comment:
What about the cases where our caller is not the test function? For example, all of the usages of `util.submit_jobs` which end up calling this function. Seems like instead of relying on the stack frames, we should get the name of the currently running test from pytest.

Reply:
Yup. That's why I have similar code in the other main codepaths, e.g., submit_jobs and full_spec.